docs/source/cpp/compute.rst

```diff
@@ -42,6 +43,16 @@ whether the inputs are integral or floating-point).
 Functions are stored in a global :class:`FunctionRegistry` where
 they can be looked up by name.
 
+Compute Initialization
+----------------------
+
+The compute library requires a call to :func:`arrow::compute::Initialize`
+in order to register the individual functions into the :class:`FunctionRegistry`, otherwise
+only the functions required for Arrow core will be available  \(1).
+
+\(1) Those are ``add``, ``array_filter``, ``array_take``, ``dictionary_encode``, ``dictionary_decode``,
+``indices_nonzero``, ``unique`` and ``value_counts``.
```

Comment:
I'm not sure we want to list those. Users should not rely on it.

Reply:
"The set of functions required for Arrow core functionality are an implementation detail of the library, and should not be considered stable." (even if we don't list them, users could find out and rely on the core set)